### ISSUE TITLE
NK Storage: stick is reconnecting while wiping out SD card

### ISSUE THREAD

**Issue by szszszsz** [Nitrokey/nitrokey-app#48]
During wiping out sd card messages 'Nitrokey connected' and 'Nitrokey disconnected' are appearing. These should not be visible until wiping out completes.

**Comment by szszszsz** [comment]
Reported before in another project under url https://github.com/Nitrokey/nitrokey-storage-firmware/issues/1

**Comment by szszszsz** [comment]
Attaching log from dmesg. There are mostly `(app) did not claim interface 2 before use` messages with timedelta. They occur during clearing SD card operation and after it - there is a need to reinsert stick to continue work.
[issue_48.txt](https://github.com/Nitrokey/nitrokey-app/files/272426/issue_48.txt)

**Comment by jans23** [comment]
The disconnect / connect messages don't appear anymore. But also the progress bar doesn't work anymore and remains at 0% because the device is "disconnected" right after start initializing the storage.

Note: Until recently the disconnect / connect problem got partially solved for me by the following change. Now it doesn't solve the issue for me anymore. However, this isn't a proper fix because the App would identify a physical removal of the device 10 seconds late. However, I post the patch here because it might help to find a better solution.

**device.cpp: int Device::checkConnection ()**
 
```
  DisconnectCounter++;
  if (1 < DisconnectCounter) // Increase counter from 1 to 10
  {
      isConnected = false;
      newConnection = true;
      return -1;
  }
```
 

**hid_libusb.c: int HID_API_EXPORT hid_get_feature_report (hid_device * dev, unsigned char* data, size_t length)**

```
(3 /* HID feature */  << 8) |
report_number, dev->interface, (unsigned char *) data, length, 1000 /* timeout millis */ ); // Increase timeout from 1000 to 10000
```

**Comment by jans23** [comment]
@szszszsz I discussed with @rudbo that this is more a low-level issue which @rudbo will take care of. @szszszsz are you working on any other related fixes you want to push?

**Comment by szszszsz** [comment]
Hi @jans23 !
While investigating I have discovered that clearing the SD card is not working on my USB 3.0 ports. Maybe this is why it was not working for you?
My PR change was only to temporarily disable tray messages to not annoy user with it, so this should not influence progress bar.
I have tried to found any clue about disconnecting, but until now with no result.

**Comment by rudbo** [comment]
SHA-1: 79abe9a6bb8db513479888a046554924c4b4cbf7
* Avoid USB fail when init SD data with random char

**Comment by jans23** [comment]
This issue has been fixes with Storage firmware v0.34.
